src/Microsoft.DotNet.Build.Tasks/Targets/versioning.targets

```diff
@@ -0,0 +1,43 @@
+<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <UsingTask TaskName="GetDailyBuildNumber" 
+    AssemblyFile="$(ToolsDir)Microsoft.DotNet.Build.Tasks.dll"
+    Condition="Exists('$(ToolsDir)Microsoft.DotNet.Build.Tasks.dll')" />
+
+  <Target Name="GetDailyBuildNumber"
+    BeforeTargets="Build"
+    Condition="'$(DailyBuildNumGenEnabled)' == 'true'"
+    >
+    <GetDailyBuildNumber
+      VersionPropsFile="$(SourceDir)version.props">
+      <Output PropertyName="DailyBuildNumber" TaskParameter="DailyBuildNumber" />
+    </GetDailyBuildNumber>
+  </Target>
+
+  <Target Name="CommitDailyBuildNumber"
+    AfterTargets="BuildPackages"
+    Condition="'$(DailyBuildNumGenEnabled)' == 'true'"
+    >
+    <!-- configure the commit to show up as the dotnet bot -->
+    <Exec
+      WorkingDirectory="$(SourceDir)"
```

Comment:
I thought about that however I feel that it obscures the implementation and makes it a bit more difficult to change (this implementation is completely transparent).  That said if there are other advantages to using the git API over this method that I'm missing I'm all for changing it (one thing that comes to mind is retry logic if the push fails, hmm...).